### PR TITLE
Added JPEGPhoto and createhomedir

### PR DESCRIPTION
**Pull request by gniemetz**
My first try to build a "real" python program, so please be patient :)

I've added the support for creating JPEGPhoto, also added support to create the home directory

Thanks you!

best regards
Gerd

### CODE REVIEW

File: createuserpkg [hunk 1]
```diff
@@ -36,16 +35,20 @@ def main():
     required_user_options = optparse.OptionGroup(
         parser, 'Required User Options')
     required_user_options.add_option(
-        '--name', '-n', help='User shortname. REQUIRED.')
+        '--name', '-n',
+        help='User shortname. REQUIRED.')
     required_user_options.add_option(
```
Comment (gregneagle) [inline createuserpkg:41]: You've made a bunch of changes unrelated to your core changes. This makes reviewing your changes harder.

File: createuserpkg [hunk 2]
```diff
@@ -119,7 +135,10 @@ def main():
         user_data['shell'] = options.shell
     if options.hidden:
         user_data['IsHidden'] = u'YES'
-
+    if options.picture:
+        from locallibs import imageutil
```
Comment (gregneagle) [inline createuserpkg]: imports really belong at the top of a Python file unless you have a really good reason to do them conditionally.

File: locallibs/imageutil.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        temp_image = os.path.join(os.path.sep, 'private/var/tmp',
+                                  os.path.splitext(os.path.basename(image_path))[0] + '.jpg')
+        # convert image_path to jpeg with 72 dpi and max. width/height 512 px
+        cmd = ('''/usr/bin/sips -s format jpeg "{0}" '''
+               '''-s formatOptions high -s dpiHeight 72 -s dpiWidth 72 --resampleHeightWidthMax 512 '''
+               '''--out "{1}"''').format(
+            image_path,
+            temp_image)
+        try:
+            subprocess.call(cmd, shell=True)
```
Comment (gregneagle) [inline locallibs/imageutil.py]: Please don't use shell=True with subprocess. Properly format your command arguments into a list and use the command list with subprocess.call.

File: locallibs/userpkg.py [hunk 4]
```diff
@@ -88,6 +89,9 @@
     /usr/bin/killall DirectoryService 2>/dev/null || /usr/bin/killall opendirectoryd 2>/dev/null
 """
 
+PI_CREATEHOMEDIR = """
+/usr/sbin/createhomedir -c
+"""
```
Comment (gregneagle) [inline locallibs/userpkg.py]: Using createhomedir will only work when a package is installed to the current boot volume. All the other options for pycreateuserpkg work no matter if the package target is the current boot volume or an alternate volume (as when installing into an AutoDMG image, or via a Imagr or DeployStudio or Bootstrappr workflow). I'm hesitant to merge a feature that works only sometimes.

**Review by gregneagle — COMMENTED** [review]
See in-line comments.

File: locallibs/userpkg.py (hunk 4 above)
Comment (poundbangbash) [reply locallibs/userpkg.py]: In the latest 10.13.4 beta 4 there is also a new behavior to createmobileaccount that prompts in the cli run for an admin account to set the SecureToken. There is a `-D` option to suppress it. There should be some logic to detect the OS and act accordingly.

-Eric

Comment (gregneagle) [reply locallibs/userpkg.py]: `createmobileaccount` and `createhomedir` are different tools; I don't know if there are similar new issues in 10.13.4 that would affect `createhomedir`.

File: locallibs/imageutil.py (hunk 3 above)
Comment (gniemetz) [reply locallibs/imageutil.py]: will do

File: locallibs/userpkg.py (hunk 4 above)
Comment (gniemetz) [reply locallibs/userpkg.py]: I will put in into the if block, under the killall statements

File: createuserpkg (hunk 2 above)
Comment (gniemetz) [reply createuserpkg]: I thought i only import it if there is a image as argument, but i will put it on top

File: createuserpkg (hunk 1 above)
Comment (gniemetz) [reply createuserpkg:41]: sorry about that, thought it would be easier to read

File: locallibs/userpkg.py [hunk 5]
```diff
@@ -88,6 +88,9 @@
     /usr/bin/killall DirectoryService 2>/dev/null || /usr/bin/killall opendirectoryd 2>/dev/null
 """
 
+PI_CREATEHOMEDIR = """
+    /usr/sbin/createhomedir -c
+"""
```
Comment (gregneagle) [inline locallibs/userpkg.py:93]: I don't think you've addressed my issue with `createhomedir` not working for anything other than installs to /.
I think the postinstall script bit needs to test $3 and do the right thing, and I think the overall `createuserpkg` tool needs to warn people in the help output that this option works only in some scenarios; etc.

Comment (gniemetz) [reply locallibs/userpkg.py:93]: I will add the createhomedir command to the pi_live_actions, so it gets written into the if [ "$3" == "/" ] block. Additionally i will add a notice to the help text that says that it only work an a boot volume.
Would that be ok?
Thanks again
Gerd

**Comment by gniemetz** [comment]
Have you had time to look at my latest pull request

Thanks in advance
regards
Gerd

**Comment by gregneagle** [comment]
Not yet, no.

**Comment by gregneagle** [comment]
Sorry, coming back to this now after a billion-year break; did you want to still work on merging this?

**Pull request closed without merging**